src/asound/bluealsa-pcm.c

```diff
@@ -1715,20 +1994,34 @@ SND_PCM_PLUGIN_DEFINE_FUNC(bluealsa) {
 		}
 	}
 
-	/* If the BT transport codec is not known (which means that PCM sample
-	 * rate is also not know), we cannot construct useful constraints. */
+	/* If the BT transport codec is not known (which means that PCM sampling
+	 * rate is also not known), we cannot construct useful constraints. */
 	if (pcm->ba_pcm.rate == 0) {
 		ret = -EAGAIN;
 		goto fail;
 	}
 
-	if (stream == SND_PCM_STREAM_CAPTURE)
+	/* HW compatible mode applies only to a2dp-sink, hfp-hf and hsp-hs. */
+	if (pcm->ba_pcm.transport & (BA_PCM_TRANSPORT_A2DP_SOURCE | BA_PCM_TRANSPORT_MASK_AG))
```

Comment:
I've just started reviewing the HW compat code and I have a question. Is this line correct? The documentation says that compatibility mode is for capture and playback as well...

> silence - Inserts silence for capture streams, or simply drops frames for playback streams [...]

Reply:
I've modified the line in question to allow silence mode on all roles, and tested with `alsaloop` on a HFP-AG node. Seems to be working well.